magenta-lib/src/main/scala/magenta/deployment_type/S3.scala

```diff
@@ -1,11 +1,37 @@
 package magenta.deployment_type
 
-import java.io.File
-
 import magenta.Datum
-import magenta.json.JValueExtractable
 import magenta.tasks.S3Upload
-import org.json4s.JsonAST._
+import play.api.data.validation.ValidationError
+import play.api.libs.json._
+
+case class PatternValue(pattern: String, value: String) {
+  lazy val regex = pattern.r
+}
+object PatternValue {
+  implicit val reads = Json.reads[PatternValue]
+
+  implicit val patternValueReads = new Reads[List[PatternValue]] {
```

Comment:
Sounds like a plan. I'll have a look at doing that.

Are you happy with removing the JValueExtractor stuff entirely? Felt like it duplicated `Reads`.